ansible_ai_connect/ai/api/views.py

```diff
@@ -189,15 +189,19 @@ def initialize_request(self, request, *args, **kwargs):
         self.event: schema1.Schema1Event = self.schema1_event()
         self.event.set_request(initialised_request)
 
-        # TODO: when we will move the request_serializer handling in this
-        # class, we will change this line below. The model_id attribute
-        # should exposed as self.validated_data.model, or using a special
-        # method.
-        # See: https://github.com/ansible/ansible-ai-connect-service/pull/1147/files#diff-ecfb6919dfd8379aafba96af7457b253e4dce528897dfe6bfc207ca2b3b2ada9R143-R151  # noqa: E501
-        self.req_model_id: str = ""
+        if hasattr(request, "data"):
+            self.load_parameters(request)
 
         return initialised_request
 
+    def load_parameters(self, request) -> Response:
+        if hasattr(self, "request_serializer_class"):
+            request_serializer = self.request_serializer_class(data=request.data)
+            request_serializer.is_valid(raise_exception=True)
+            self.validated_data = request_serializer.validated_data
+            if req_model_id := self.validated_data.get("mode_id"):
```

Comment:
`model_id`

Reply:
Since we have different `model_id` ( or `modelName` ) variables, I use `req_model_id` internally to store the value the user requested in the payload.

Reply:
Yes, but you've typed `mode_id` not `model_id`.

Reply:
oh you mean `model_id`, not `mode_id` :D. Thanks!